pkg/client/client.go

```diff
@@ -102,9 +102,10 @@ type Client struct {
 	lifecycleExecutor   LifecycleExecutor
 	buildpackDownloader BuildpackDownloader
 
-	experimental    bool
-	registryMirrors map[string]string
-	version         string
+	experimental       bool
+	registryMirrors    map[string]string
+	version            string
+	insecureRegistries []string
```

Comment:
I think we are not using the `insecureRegistries` values when fetching images. In theory the values configured here must be use during the `Fetch` operation.

During the [Fetch](https://github.com/buildpacks/pack/blob/8ed450d44de15cefe5875c22ae87293cc625e69d/pkg/image/fetcher.go#L86) operation we can try to access images on different sources:
- Daemon
- Registry
- On disk in OCI layout format
The `insecureRegistries` make sense for remote registry, right? we need to update our remote fetching with the correct configuration from imgutil.

```go

func (f *Fetcher) fetchRemoteImage(name string) (imgutil.Image, error) {
       // maybe here, we need to check if we have insecureRegistries and for each of them
       // we need to create a options with 
       // remote.WithRegistrySetting(<the insecure registry>, true)
       // passthrough the options  to the creation of the image

	image, err := remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
	if err != nil {
		return nil, err
	}

	if !image.Found() {
		return nil, errors.Wrapf(ErrNotFound, "image %s does not exist in registry", style.Symbol(name))
	}

	return image, nil
}

```

Reply:
Done!